docs/roadmap.md

```diff
@@ -0,0 +1,177 @@
+  * File System
+  * DDP
+* Mock partial or full response
+* Route operations to alternate endpoints
+  * Partial operations to REST endpoint
+  * Partial operations to GraphQL endpoint
+  * Partial operations to Mock
+* Split single operation into multiple operations
+  * Race one operation to local storage and another to network
+* Per request retries based on network status or response
+* Polling
+* Batching at the transport/fetch layer
+  * Perform batch operations within a composed link
+* Deduplicate of on-the-wire requests
+* Re-route and modify errors
+  * transform network error into GraphQL error
+
+<br>
+
+* Network layer caching
+  * Denormalized
+  * Normalized
+  * Local Storage
+  * IndexDB
+
+If you think of more use-cases, please open a PR adding to this list and make sure to clearly explain your use-case.
+
+## Open questions, discussion
+
+Throughout the design process, a couple questions have surfaced that may prompt conversation.
+If you want to see something here, please open a PR with your proposed change or an Issue for discussion.
+
+<!--
+### How should we pass context to `next` callback along with the query data?
+
+Should we pass the context from the response (for example relevant HTTP headers or status code) to the `next` callback of the observable like this:
+
+```js
+next({
+  data,
+  errors,
+  extensions,
+  context
+})
+```
+
+or like this:
+
+```js
+next({
+  result: {
+    data,
+    errors,
+    extensions
+  },
+  context
+})
+```
+-->
+
+### Should we include `status()` the Link interface that could contain user defined data and an enum for state?
+
+```js
+enum State {
+  cold
+  started
+  stopped
+  errored
+  completed
+}
+```
+
+If `status` exists and an `Observable` has terminated, should `subscribe` throw an error or call complete immediately, avoiding memory leaks.
+
+<!--
+### Does the subscribe method need to be overloaded? -> Yes, to make the Observable compatible
+
+Currently the `subscribe` method has two different signatures, one that takes three functions `next`, `error`, and `complete` and another that takes a `Subscriber`.
+Positional arguments are more prone to errors than passing object containing arguments.
+The three function signature is present for compatibility with GraphiQL.
+
+The suggested behavior of dealing with the overload is found in `AbstractObservable`.
+-->
+
+### Which convenience functions would you like?
+
+The current functions provided:
+
+* ApolloLink
+  * asPromiseWrapper
+
+Proposed additions include:
+
+* Pull information from the Operation as part of `Operation` or `LinkUtil` library
+  * hasQuery
+  * hasMutation
+  * hasSubscription
+  * getQuery
+  * getMutation
+  * getSubscription
+  * annotation support?
+
+<!--
+* Observable additions as required or optional part of interface and implemented in `AbstractObservable`
+  * `map`
+  * `filter`
+  * `catch`
+  * `finally`
+-->
+
+### Should GraphQL errors be propagated up the stack with `next` or `error`?
```

Comment:
I think this distinction is great since a graphql error can also return data. Previously we couldn't handle this situation, now we can